hazelcast/src/main/java/com/hazelcast/map/impl/operation/BasePutOperation.java

```diff
@@ -37,8 +37,7 @@ public BasePutOperation() {
 
     @Override
     protected void afterRunInternal() {
-        Object value = isPostProcessing(recordStore)
-                ? recordStore.getRecord(dataKey).getValue() : dataValue;
+        Object value = getValueForPublishing();
```

Comment:
@ahmetmircik can this lead to unexpected sharing of instances because an object isn't cloned but the same instance is potentially shared between multiple threads? The complex flow in map operations obscures what is going on

Reply:
This change only makes local thread's processing more efficient by doing less deserialization. Other things are same with what we have currently.